edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/ConfigUpdater.cs

```diff
@@ -129,6 +133,7 @@ void UpdateStoreAndForwardConfig(StoreAndForwardConfiguration storeAndForwardCon
             if (storeAndForwardConfiguration != null)
             {
                 this.messageStore?.SetTimeToLive(TimeSpan.FromSeconds(storeAndForwardConfiguration.TimeToLiveSecs));
+                storeAndForwardConfiguration.StoreLimits.ForEach(b => this.storageSpaceChecker.SetMaxSizeBytes(Option.Maybe<long>(b.MaxSize)));
```

Comment:
This won't disable the DiskSpaceChecker if StoreLimits is Option.None as ForEach is only called when StoreLimits has a value.